compliance_checker/ioos.py

```diff
@@ -346,6 +348,353 @@ def check_units(self, ds):
+            'standard_name_vocabulary',
+            'summary',
+            'title'
+        ]
+
+        self.rec_atts = [
+            'contributor_email',
+            'contributor_name',
+            'contributor_role',
+            'contributor_role_vocabulary',
+            'contributor_url',
+            'creator_address',
+            'creator_city',
+            'creator_name',
+            'creator_phone',
+            'creator_postalcode',
+            'creator_state',
+            'creator_type',
+            'institution',
+            'instrument',
+            'ioos_ingest',
+            'keywords',
+            'platform_id',
+            'publisher_address',
+            'publisher_city',
+            'publisher_name',
+            'publisher_phone',
+            'publisher_postalcode',
+            'publisher_state',
+            'publisher_type',
+            'references',
+        ]
+
+    @check_has(BaseCheck.HIGH)
+    def check_high(self, ds):
+        '''
+        Performs a check on each highly recommended attributes' existence in the dataset
+
+        :param netCDF4.Dataset ds: An open netCDF dataset
+        '''
+        return self.required_atts
+
+    @check_has(BaseCheck.MEDIUM)
+    def check_recommended(self, ds):
+        '''
+        Performs a check on each recommended attributes' existence in the dataset
+
+        :param netCDF4.Dataset ds: An open netCDF dataset
+        '''
+        return self.rec_atts
+
+    def check_standard_name(self, ds):
+        """
+        Wrapper for checking standard names using the CF module.
+        Extends the StandardNameTable to include QARTOD variable
+        standard names.
+        """
+
+        return self.cf1_7.check_standard_name(ds)
+
+    def check_units(self, ds):
+        """
+        Wrapper to check units with the CF module.
+        """
+
+        return self.cf1_7.check_units(ds)
+
+    def check_vars_have_attrs(self, ds):
+        """
+        Using the tuples defined in __init__, check that each variable has
+        the attributes using the corresponding priority.
+
+        :param netCDF4.Dataset: open netCDF4 dataset
+        """
+
+        results = []
+        # NOTE should it also find 'geospatial` variables?
+        for geo_var in get_geophysical_variables(ds):
+            for attr_tuple in self.check_var_attrs:
+                results.append(
+                    self._has_var_attr(
+                        ds,
+                        geo_var,
+                        attr_tuple[0], # attribute name
+                        attr_tuple[0], # attribute name used as 'concept_name'
+                        attr_tuple[1]  # priority level
+                    )
+                )
+
+        return results
+
+
+    def check_single_platform(self, ds):
+        """
+        Verify that a dataset only has a single platform attribute.
+
+        Args:
+            ds (netCDF-4 Dataset): open Dataset object
+
+        Returns:
+            None
```

Comment:
This function appears to return a `Result`, so this should be reflected in the docstring.